src/service/HomeService.php

```diff
@@ -79,7 +82,16 @@ public function getLastPosts(): array
 
         $results = $this->_postRepository->getListOfPosts();
 
-        return $this->_postsMapper->transformToListOfPostModel($results);
+        $postModels = [];
+
+        foreach ($results as $entity) {
```

Comment:
il y a une autre manière de faire. Tu peux utiliser la fonction array_map de PHP. C'est un coup de main à prendre pour comprendre les closure (fonction anonyme).

```suggestion
        $postModels = array_map(function ($entity) {
               $username = $this->_userRepository->getUsername($entity->getIdUser());
                return $this->_postsMapper->transformToPostModel($entity, $username);
         }, $results);

```

Reply:
fait